Copyright (c) 2015 Microsemi Corporation

Permission is hereby granted, free of charge, to any person obtaining a copy
of this software and associated documentation files (the "Software"), to deal
in the Software without restriction, including without limitation the rights
to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
copies of the Software, and to permit persons to whom the Software is
furnished to do so, subject to the following conditions:

The above copyright notice and this permission notice shall be included in
all copies or substantial portions of the Software.

THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
THE SOFTWARE.


[PATCH] BZ#19367: If libupnp receives numerous SSDP M-Search packets in short time period,
        libupnp cannot handle all packets immediately, this mechanism is to prevent
        this kind of attack. When buffer queue is avaliable to handle new requests,
        UPnP behavior will back to normal. During unavaliable period, console output
        will be full of this message, so suppress it.
Signed-off-by: Palle Christensen <palle.christensen@microsemi.com>
Index: b/threadutil/src/ThreadPool.c
===================================================================
--- a/threadutil/src/ThreadPool.c
+++ b/threadutil/src/ThreadPool.c
@@ -827,7 +827,16 @@
 
 	totalJobs = tp->highJobQ.size + tp->lowJobQ.size + tp->medJobQ.size;
 	if (totalJobs >= tp->attr.maxJobsTotal) {
-		fprintf(stderr, "total jobs = %ld, too many jobs", totalJobs);
+        /*
+            BZ#19367: If libupnp receives numerous SSDP M-Search packets in short time period,
+                libupnp cannot handle all packets immediately, this mechanism is to prevent
+                this kind of attack. When buffer queue is avaliable to handle new requests,
+                UPnP behavior will back to normal. During unavaliable period, console output
+                will be full of this message, so suppress it.
+        */
+#ifdef __NOTDEF__
+        fprintf(stderr, "total jobs = %ld, too many jobs", totalJobs);
+#endif
 		goto exit_function;
 	}
 	if (!jobId)
